opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -46,46 +45,46 @@ option go_package = "github.com/open-telemetry/opentelemetry-proto/gen/go/metric
+//             |  .  |   |+-----+                 |
+//             |  .  |   +------------------------+
 //             |  .  |                   .
 //             |  .  |                   .
 //             |  .  |                   .
-//             |  .  |   +-----------------------------------+
-//             |  .  |   |TimeSeries N                       |
-//             +-----+   |+-----+-----+   +-----+            |
-//             |  N  |-->||label|label|...|label|            |
-//             +-----+   |+-----+-----+   +-----+            |
-//                       |+-----+-----+-----+-----+   +-----+|
-//                       ||value|value|value|value|...|value||
-//                       |+-----+-----+-----+-----+   +-----+|
-//                       +-----------------------------------+
+//             |  .  |   +------------------------+
+//             |  .  |   |DataPoint M             |
+//             +-----+   |+-----+-----+   +-----+ |
+//             |  M  |-->||label|label|...|label| |
+//             +-----+   |+-----+-----+   +-----+ |
+//                       |+-----+                 |
+//                       ||value|                 |
+//                       |+-----+                 |
+//                       +------------------------+
 //
 //-----------------------------------------------------------------------
-// TimeSeries contain an array of values, each value corresponding to a given moment in
-// time. Each element of TimeSeries array is timestamped.
+// DataPoint is a value of specific type corresponding to a given moment in
+// time. Each DataPoint is timestamped.
 //
-// TimeSeries are strongly typed: the element of a TimeSeries array has
-// a specific ProtoBuf message depending on the value type of the metric and thus
-// there are currently 4 TimeSeries messages, which correspond to the types of metric values.
+// DataPoint is strongly typed: each DataPoint type has a specific Protobuf message
+// depending on the value type of the metric and thus there are currently 4 DataPoint
+// messages, which correspond to the types of metric values.
 message Metric {
   // metric_descriptor describes the Metric.
   MetricDescriptor metric_descriptor = 1;
 
-  // data is a list of one or more TimeSeries for a single metric, where each timeseries has
-  // one or more points. Only one of the following fields is used for the data, depending on
-  // the type of the metric defined by MetricDescriptor.type field.
-  repeated Int64TimeSeries int64_timeseries = 2;
-  repeated DoubleTimeSeries double_timeseries = 3;
-  repeated HistogramTimeSeries histogram_timeseries = 4;
-  repeated SummaryTimeSeries summary_timeseries = 5;
+  // Data is a list of one or more DataPoints for a single metric. Only one of the
+  // following fields is used for the data, depending on the type of the metric defined
+  // by MetricDescriptor.type field.
+  repeated Int64DataPoint int64_datapoints = 2;
```

Comment:
if I am reading this right, it means for every measurement point recorded we need to send all the labels, even though a time series is defined as data points with identical labels, which is the most prevalent use case (especially when using handle/bound instruments).

Reply:
@yurishkuro You are reading it right. See discussion on https://github.com/open-telemetry/opentelemetry-proto/issues/54

Can you describe a use case which you believe is prevalent where there is a need to transmit multiple data points with the same set of label values in one request? Or are saying that we could avoid transmitting the labels at all if they didn't change (that would be some sort of streaming or history-aware protocol, which this is not).

If there is indeed a good use-case for multiple data points this PR should not be merged.

Reply:
@yurishkuro not for every measurement. At the bound level we will aggregate for a period of time (e.g. 10 sec) then export one point that represents the current aggregated value at that bound.

Reply:
Hmm, right, what Bogdan said. I believe SDK is going to do collection cycles periodically and only export one data point per labelset per cycle.

Reply:
I remember we had a discussion some time ago about raw measures vs. instruments, with one example someone mentioned was that even a counter can be treated as a raw measure if we treat each delta as an independent event. Are we discounting the possibility of ever exporting these raw measures without any aggregation on the client side?

In any case, this is not a blocking issue for me.

cc @robskillington

Reply:
PS I do agree with the proposal in #54 that for such special case of multi-valued metrics we could have a different API / data structure.

Reply:
>Are we discounting the possibility of ever exporting these raw measures without any aggregation on the client side?

It is still possible using this schema, though it will be less efficient (label values will be transmitted for every data point). Also, you are right, we have the option to have a different request type / data structure that is more optimal for that case.